connection.go

```diff
@@ -122,83 +118,77 @@ func (c *Conn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, e
 		return nil, errors.Join(errBeginTx, errIsolationLevelNotSupported)
 	}
 
-	if _, err := c.ExecContext(ctx, `BEGIN TRANSACTION`, nil); err != nil {
+	if _, err := conn.ExecContext(ctx, `BEGIN TRANSACTION`, nil); err != nil {
 		return nil, err
 	}
 
-	c.tx = true
-	return &tx{c}, nil
+	conn.tx = true
+	return &tx{conn}, nil
 }
 
 // Close closes the connection to the database.
 // It implements the driver.Conn interface.
-func (c *Conn) Close() error {
-	if c.closed {
+func (conn *Conn) Close() error {
+	if conn.closed {
 		return errClosedCon
 	}
-	c.closed = true
-	C.duckdb_disconnect(&c.duckdbCon)
+	conn.closed = true
+	m.Disconnect(&conn.conn)
 	return nil
 }
 
-func (c *Conn) extractStmts(query string) (C.duckdb_extracted_statements, C.idx_t, error) {
-	cQuery := C.CString(query)
-	defer C.duckdb_free(unsafe.Pointer(cQuery))
-
-	var stmts C.duckdb_extracted_statements
-	count := C.duckdb_extract_statements(c.duckdbCon, cQuery, &stmts)
+func (conn *Conn) extractStmts(query string) (m.ExtractedStatements, m.IdxT, error) {
+	var extractedStmts m.ExtractedStatements
 
+	count := m.ExtractStatements(conn.conn, query, &extractedStmts)
 	if count == 0 {
-		errMsg := C.GoString(C.duckdb_extract_statements_error(stmts))
-		C.duckdb_destroy_extracted(&stmts)
+		errMsg := m.ExtractStatementsError(extractedStmts)
+		m.DestroyExtracted(&extractedStmts)
 		if errMsg != "" {
-			return nil, 0, getDuckDBError(errMsg)
+			return extractedStmts, 0, getDuckDBError(errMsg)
```

Comment:
nit: Even as `extractedStmts` is nil in the error case, I would return nil to be explicit. The same would apply for line 150.